code/modules/surgery/surgery.dm

```diff
@@ -27,19 +18,47 @@
 	return 1
 
 
-/datum/surgery/proc/next_step(mob/user, mob/living/carbon/target)
-	if(step_in_progress)	return
-
-	var/datum/surgery_step/S = get_surgery_step()
+/datum/surgery/proc/next_step(mob/user, mob/living/carbon/target, obj/item/tool)
+	if(step_in_progress)	return FALSE
+	. = TRUE // Person 
+	var/list/steps = get_surgery_steps(user, target, tool)
+	var/datum/surgery_step/S
+	if(steps.len == 0)
+		return FALSE // No surgery steps. So stab that person in the ****
+	if(steps.len == 1)
+		S = steps[steps[1]]
+	else
+		var/P = input("Begin which procedure?", "Surgery", null, null) as null|anything in steps
+		if(P && user && user.Adjacent(target) && tool == user.get_active_hand() && can_operate(target))
+			S = steps[P]
 	if(S)
-		if(S.try_op(user, target, user.zone_selected, user.get_active_hand(), src))
-			return 1
-	return 0
-
-/datum/surgery/proc/get_surgery_step()
-	var/step_type = steps[status]
-	return new step_type
-
+		var/result
+		. = TRUE
+		do
```

Comment:
Don't think I've ever seen a `do...while` in this codebase before 🤔

Reply:
I avoid them if I can, seems like a weird operation to me

Reply:
It is used in some places. I can hack it out and set result to SURGERY_CONTINUE
But Id rather have it like this. It will always do the loop once. Then it'll act like a normal while loop